File: app/src/main/java/net/feheren_fekete/applist/applistpage/model/AppData.java
package net.feheren_fekete.applist.applistpage.model;

public class AppData extends StartableData {
    private String mPackageName;
    private String mClassName;

    public AppData(long id, String packageName, String className, String appName) {
        super(id, appName);
        mPackageName = packageName;
        mClassName = className;
    }

    @Override
    public String getPackageName() {
        return mPackageName;
    }

    public String getClassName() {
        return mClassName;
    }

    @Override
    public boolean equals(Object o) {
        if (o == null || !(o instanceof AppData)) {
            return false;
        }
        AppData other = (AppData) o;
        return mPackageName.equals(other.getPackageName())
                && mClassName.equals(other.getClassName());
    }

}

File: app/src/main/java/net/feheren_fekete/applist/applistpage/model/AppData.kt
package net.feheren_fekete.applist.applistpage.model

class AppData(id: Long,
              packageName: String,
              val className: String,
              val versionCode: Long,
              appName: String,
              customName: String) : StartableData(id, packageName, appName, customName) {

    override fun equals(other: Any?): Boolean {
        if (other == null || other !is AppData) {
            return false
        }
        return packageName == other.packageName && className == other.className
    }

}
